src/App.js

```diff
@@ -1,75 +1,157 @@
 import React from 'react';
+import NewTodo from './Components/NewTodo/NewTodo';
+import TodoList from './Components/TodoList/TodoList';
+import TodosFilter from './Components/TodosFilter/TodosFilter';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+class App extends React.Component {
+  state = {
+    todos: [],
+    originalTodos: [],
+    buttonSelected: 'all',
+  }
 
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-        />
-      </header>
-
-      <section className="main" style={{ display: 'block' }}>
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-
-        <ul className="todo-list">
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">sdfsdfsdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">sakgjdfgkhjasgdhjfhs</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">sddfgdfgdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-        </ul>
-      </section>
+  handleAddNewTodo = (todo) => {
+    if (todo.title.length > 0) {
+      this.setState(prevState => ({
+        todos: [...prevState.todos, todo],
+        originalTodos: [...prevState.originalTodos, todo],
+      }));
+    }
+
+    this.handleButtonChange(this.state.buttonSelected);
+  }
+
+  handleDoubleClickEditTitle = (filmTitle, todoId) => {
+    this.setState(prevState => ({
+      todos: [...prevState.todos]
+        .map(todo => (
+          todo.id === todoId
+            ? { ...todo, title: filmTitle }
+            : todo
+        )),
+      originalTodos: [...prevState.originalTodos]
+        .map(orTodo => (
+          orTodo.id === todoId
+            ? { ...orTodo, title: filmTitle }
+            : orTodo
+        )),
+    }));
+  }
+
+  handleDeleteTodo = (todoId) => {
+    this.setState(prevState => ({
+      todos: [...prevState.todos]
+        .filter(todo => (todo.id !== todoId)),
+      originalTodos: [...prevState.originalTodos]
+        . filter(todo => (todo.id !== todoId)),
```

Comment:
```suggestion
        .filter(todo => (todo.id !== todoId)),
```
кстати, фильтер тоже не меняет исходный массив